ORM/DataObject.php

```diff
@@ -1682,8 +1700,7 @@ public function getRelationType($component) {
 	 * @throws InvalidArgumentException
 	 */
 	public function inferReciprocalComponent($remoteClass, $remoteRelation) {
-		/** @var DataObject $remote */
-		$remote = $remoteClass::singleton();
+		$remote = DataObject::singleton($remoteClass);
```

Comment:
DataObject::singleton() simple invokes Injector::inst()->get(), but it has `@return static` to enable type-hinting, whereas Injector::inst()->get() has none. I've found that typehinting in code helps uncover bugs a lot easier, so would you agree it's beneficial to leave it?